cpp/src/gis/point_in_polygon.cu

```diff
@@ -0,0 +1,186 @@
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+        }
+        if ((count > 0) && (count % 2 == 0)) point_is_in_polygon[start_idx] = 1;
+        else point_is_in_polygon[start_idx] = 0;
+    }
+}
+
+struct point_in_polygon_functor {
+    template <typename col_type>
+    static constexpr bool is_supported()
+    {
+        return std::is_arithmetic<col_type>::value ;
+    }
+ 
+    template <typename col_type, typename std::enable_if< is_supported<col_type>() >::type* = nullptr>
+    void operator()(const gdf_column* d_poly_lats,
+                    const gdf_column* d_poly_lons,
+                    const gdf_column* d_point_lats,
+                    const gdf_column* d_point_lons,
+                    gdf_column* d_point_is_in_polygon)
+    {
+
+        gdf_size_type min_grid_size = 0, block_size = 0;
+        CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, point_in_polygon_kernel<col_type>) );
+
+        point_in_polygon_kernel<col_type> <<< min_grid_size, block_size >>> (static_cast<col_type*>(d_poly_lats->data), 
+                static_cast<col_type*>(d_poly_lons->data), static_cast<col_type*>(d_point_lats->data), static_cast<col_type*>(d_point_lons->data), 
+                d_poly_lats->size, d_point_lats->size, static_cast<int8_t*>(d_point_is_in_polygon->data) );
+    }
+
+    template <typename col_type, typename std::enable_if< !is_supported<col_type>() >::type* = nullptr>
+    void operator()(const gdf_column* d_poly_lats,
+                    const gdf_column* d_poly_lons,
+                    const gdf_column* d_point_lats,
+                    const gdf_column* d_point_lons,
+                    gdf_column* d_point_is_in_polygon)
+    {
+        CUDF_FAIL("Non-arithmetic operation is not supported");
+    }
+};
+    
+}   //  namespace
+
+namespace cudf {
+namespace gis {
+
+gdf_column* point_in_polygon(const gdf_column* polygon_lats, const gdf_column* polygon_lons, const gdf_column* point_lats, const gdf_column* point_lons)
+{
+    cudaStream_t stream;
+    CUDA_TRY( cudaStreamCreate(&stream) );
+        
+    CUDF_EXPECTS(polygon_lats != nullptr && polygon_lons != nullptr, "polygon data cannot be empty");
+    CUDF_EXPECTS(point_lats != nullptr && point_lons != nullptr, "query points are empty");
+    CUDF_EXPECTS(polygon_lats->size == polygon_lons->size, "polygon size doesn't match");
+    CUDF_EXPECTS(point_lats->size == point_lons->size, "query points size doesn't match");
+    CUDF_EXPECTS(polygon_lats->dtype == polygon_lons->dtype, "polygon type doesn't match");
+    CUDF_EXPECTS(polygon_lats->dtype == point_lats->dtype, "type of query points and polygon doesn't match");
+    CUDF_EXPECTS(point_lats->dtype == point_lons->dtype, "query points type doesn't match");
```

Comment:
You also need to check that point_lats->dtype == point_lons->dtype.

Reply:
It's already added in line 149. I applied a kind of transitive relation.